packages/cli/src/commands/eject/index.ts

```diff
@@ -0,0 +1,196 @@
+        ],
+      },
+    });
+
+    return 0;
+  }
+
+  try {
+    return _eject(flags);
+  } catch (error) {
+    console.error(`${errorLabel()} Command failed`);
+
+    if (error.stack) {
+      console.error(`\n${error.stack}`);
+    }
+
+    process.exit(1);
+  }
+}
+
+async function _eject(flags: EjectOptions) {
+  let folderPath = flags._[1] !== undefined ? String(flags._[1]) : undefined;
+
+  if (folderPath === undefined) {
+    folderPath = process.cwd();
+  } else {
+    folderPath = path.resolve(process.cwd(), folderPath);
+  }
+
+  /**
+   * First we make sure that the destination has the correct files
+   * and that there won't be any files overwritten in the process.
+   *
+   * If there are any and `force` was not specified we abort.
+   */
+  const { astroConfigPath, srcPath, pkgJsonPath, astroIntegrationPath, srcDestPath } = validateDestination(
+    folderPath,
+    flags.force,
+  );
+
+  /**
+   * We proceed with the astro configuration.
+   *
+   * There we must disable the default routes so that the
+   * new routes that we're copying will be automatically picked up.
+   */
+  const astroConfig = await parseAstroConfig(astroConfigPath);
+
+  replaceArgs({ defaultRoutes: false }, astroConfig);
+
+  fs.writeFileSync(astroConfigPath, generateAstroConfig(astroConfig));
+
+  // we copy all assets from the `default` folder into the `src` folder
+  fs.cpSync(srcPath, srcDestPath, { recursive: true });
+
+  /**
+   * Last, we ensure that the `package.json` contains the extra dependencies.
+   * If any are missing we suggest to install the new dependencies.
+   */
+  const pkgJsonContent = fs.readFileSync(pkgJsonPath, 'utf-8');
+  const indent = detectIndent(pkgJsonContent).indent || '  ';
+  const pkgJson: PackageJson = JSON.parse(pkgJsonContent);
+
+  const astroIntegrationPkgJson: PackageJson = JSON.parse(
+    fs.readFileSync(path.join(astroIntegrationPath, 'package.json'), 'utf-8'),
+  );
+
+  const newDependencies = [];
+
+  for (const dep of REQUIRED_DEPENDENCIES) {
+    if (!(dep in pkgJson.dependencies) && !(dep in pkgJson.devDependencies)) {
+      pkgJson.dependencies[dep] = astroIntegrationPkgJson.dependencies[dep];
+
+      newDependencies.push(dep);
+    }
+  }
+
+  updateWorkspaceVersions(pkgJson.dependencies, TUTORIALKIT_VERSION, (dependency) =>
+    REQUIRED_DEPENDENCIES.includes(dependency),
+  );
+
+  if (newDependencies.length > 0) {
+    fs.writeFileSync(pkgJsonPath, JSON.stringify(pkgJson, undefined, indent), { encoding: 'utf-8' });
+
+    console.log(
+      primaryLabel('INFO'),
+      `New dependencies added: ${newDependencies.join(', ')}. Install the new dependencies before proceeding`,
+    );
+
+    if (!flags.defaults) {
+      const packageManager = (await preferredPM(pkgJsonPath)).name;
+
+      const answer = await prompts.confirm({
+        message: `Do you want to install those dependencies now using ${packageManager}?`,
```

Comment:
Nit: Shall we colorize the preferred package manager?

Reply:
Oh good idea! I'll play around with everything we have so far to see which color would be the best :+1: